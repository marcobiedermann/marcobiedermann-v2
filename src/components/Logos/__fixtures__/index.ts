import logo from '../../Logo/__fixtures__';

const logos = [logo];

export default logos;
