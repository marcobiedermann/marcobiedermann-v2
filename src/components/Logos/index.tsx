import classNames from 'classnames';
import React, { FC } from 'react';
import Logo, { LogoProps } from '../Logo';
import styles from './style.module.css';

export interface LogosProps {
  className?: string;
  logos: LogoProps[];
}

export const Logos: FC<LogosProps> = (props) => {
  const { className, logos, ...otherProps } = props;

  return (
    <ul className={classNames(className, styles.logos)} {...otherProps}>
      {logos.map((logo) => (
        <li key={logo.id}>
          <Logo {...logo} />
        </li>
      ))}
    </ul>
  );
};

export default Logos;
