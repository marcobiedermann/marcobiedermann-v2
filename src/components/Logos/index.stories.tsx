/* eslint-disable import/no-extraneous-dependencies */

import { storiesOf } from '@storybook/react';
import React from 'react';
import Logos from '.';
import logos from './__fixtures__';

storiesOf('Logos', module).add('default', () => <Logos logos={logos} />);
