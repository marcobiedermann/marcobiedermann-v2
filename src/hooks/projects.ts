import { graphql, useStaticQuery } from 'gatsby';
import { ProjectProps } from '../components/Project';

export interface ProjectsPageQuery {
  allContentfulProject: {
    projects: [
      {
        next: {
          id: string;
          slug: string;
        };
        node: ProjectProps;
        previous: {
          id: string;
          slug: string;
        };
      },
    ];
    pageInfo: {
      currentPage: number;
      hasNextPage: boolean;
      hasPreviousPage: boolean;
      pageCount: number;
    };
  };
}

export const useProjectsPage = (): ProjectsPageQuery => {
  return useStaticQuery<ProjectsPageQuery>(
    graphql`
      query {
        allContentfulProject(sort: { fields: [date], order: DESC }) {
          projects: edges {
            next {
              id
              slug
            }
            node {
              id
              slug
              title
              thumbnail {
                fluid {
                  ...GatsbyContentfulFluid_withWebp_noBase64
                }
              }
            }
            previous {
              id
              slug
            }
          }
        }
      }
    `,
  );
};
