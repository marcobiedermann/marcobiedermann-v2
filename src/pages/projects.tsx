import { PageProps } from 'gatsby';
import React from 'react';
import Grid from '../components/Grid';
import Layout from '../components/Layout';
import Projects from '../components/Projects';
import Section from '../components/Section';
import { useProjectsPage } from '../hooks/projects';

const BlogPage: React.FC<PageProps> = () => {
  const {
    allContentfulProject: { projects },
  } = useProjectsPage();

  return (
    <Layout>
      <Section>
        <Grid>
          <Projects projects={projects.map((project) => project.node)} />
        </Grid>
      </Section>
    </Layout>
  );
};

export default BlogPage;
