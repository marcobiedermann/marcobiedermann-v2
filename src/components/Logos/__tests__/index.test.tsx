/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Logos from '..';
import projects from '../__fixtures__';

describe('Logos component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Logos logos={projects} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
