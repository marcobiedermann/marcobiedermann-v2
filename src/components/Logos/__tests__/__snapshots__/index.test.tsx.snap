// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logos component should render correctly 1`] = `
<DocumentFragment>
  <ul
    class="logos"
  >
    <li>
      <a
        class="logo"
        href="/projects/undefined"
      >
        <img
          alt="Logo Title"
          src="https://via.placeholder.com/150x50"
        />
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
